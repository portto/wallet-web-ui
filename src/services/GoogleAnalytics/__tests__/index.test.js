// import { sendPageView } from '../';

describe("ga", () => {
  it("should initial ga and add plugins", async () => {
    jest.resetModules();
    window.location.host = "localhost:3000";
    window.ga = jest.fn();
    const { loadGA } = require.requireActual("../");

    await loadGA();

    expect(window.ga.mock.calls).toMatchSnapshot();
  });

  // it('should send default path page view with sendPageView', () => {
  //   window.ga = jest.fn();

  //   sendPageView();

  //   expect(window.ga).toHaveBeenCalledWith('send', 'pageview', location.pathname);
  // });

  // it('should send page view of path with sendPageView', () => {
  //   window.ga = jest.fn();

  //   const mockedPath = Symbol('pathname');

  //   sendPageView(mockedPath);

  //   expect(window.ga).toHaveBeenCalledWith('send', 'pageview', mockedPath);
  // });

  it("should read correct UA depends on environment", () => {
    jest.resetModules();
    window.location.host = "cobinhood.com";
    const { GA_UA } = require.requireActual("../");

    expect(GA_UA).toMatchSnapshot();
  });
});
