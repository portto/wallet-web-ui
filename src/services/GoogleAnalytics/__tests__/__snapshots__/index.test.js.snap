// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ga should initial ga and add plugins 1`] = `
Array [
  Array [
    "create",
    "UA-145395677-10",
    "auto",
  ],
]
`;

exports[`ga should read correct UA depends on environment 1`] = `"UA-145395677-10"`;
